app/controllers/api/v1/product_details_controller.rb

```diff
@@ -0,0 +1,47 @@
+class Api::V1::ProductDetailsController < ApplicationController
+    before_action :set_product_detail, only: [:show, :update, :destroy]
+    def index
+        @product_details = ProductDetail.all
+        render json: @product_details, each_serializer: ProductDetailSerializer, status: :ok
+    end
+
+    def show
+        render json: @product_detail, serialize: ProductDetailSerializer, status: :ok
+    end
+
+  def create
+    @product_detail = ProductDetail.new(product_detail_params)
+
+    if @product_detail.save
+        render json: @product_detail, status: 201
+    else
+        render_errors(@product_detail)
```

Comment:
Is this a built-in function for rails? if it is, does it through the error code to the front-end?

Reply:
yes, it throws the error in postman.